// @react-pakistan/react-commons-collection/iframe/iframe.js
"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Iframe = void 0;
var util_functions_1 = require("@react-pakistan/util-functions");
var styled_components_1 = __importDefault(require("styled-components"));
exports.Iframe = styled_components_1.default.iframe.attrs(function (_a) {
    var height = _a.height, name = _a.name, sandbox = _a.sandbox, src = _a.src, width = _a.width;
    return ({
        height: height,
        name: name || '',
        sandbox: sandbox,
        src: src,
        width: width,
    });
})(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n\n  ", ";\n"], ["\n\n  ",
    ";\n"])), function (_a) {
    var desktop = _a.desktop, laptop = _a.laptop, laptopL = _a.laptopL, mobileL = _a.mobileL, mobileM = _a.mobileM, mobileS = _a.mobileS, tablet = _a.tablet;
    return util_functions_1.responsiveComponent({
        desktop: desktop,
        laptop: laptop,
        laptopL: laptopL,
        mobileL: mobileL,
        mobileM: mobileM,
        mobileS: mobileS,
        tablet: tablet,
    });
});
var templateObject_1;
