// @react-pakistan/react-commons-collection/text-area/text-area.js
"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TextAreaWithFieldLabel = exports.TextArea = void 0;
var util_functions_1 = require("@react-pakistan/util-functions");
var styled_components_1 = __importDefault(require("styled-components"));
var field_label_1 = require("../hoc/field-label");
var type_1 = require("./type");
exports.TextArea = styled_components_1.default.textarea.attrs(function (_a) {
    var autoComplete = _a.autoComplete, autoFocus = _a.autoFocus, cols = _a.cols, dirName = _a.dirName, disabled = _a.disabled, maxLength = _a.maxLength, minLength = _a.minLength, placeholder = _a.placeholder, readOnly = _a.readOnly, required = _a.required, rows = _a.rows, value = _a.value, wrap = _a.wrap;
    return ({
        autoComplete: autoComplete || 'off',
        autoFocus: autoFocus || false,
        cols: cols || 20,
        dirName: dirName || '',
        disabled: disabled || false,
        maxLength: maxLength || 0,
        minLength: minLength || 0,
        placeholder: placeholder || '',
        readOnly: readOnly || false,
        required: required || false,
        rows: rows || 5,
        value: value,
        wrap: wrap || type_1.WRAP_TYPE.HARD,
    });
})(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n  background-color: ", ";\n  border: ", ";\n  border-radius: ", ";\n  color: ", ";\n  display: inline-block;\n  font-family: ", ";\n  font-size: ", ";\n  font-stretch: normal;\n  font-style: normal;\n  font-weight: ", ";\n  letter-spacing: ", ";\n  line-height: ", ";\n  margin: 0;\n  overflow: auto;\n  padding: 1em;\n  resize: none;\n  width: 100%;\n\n  &:focus {\n    outline: none;\n    border: ", ";\n  }\n\n  &::-webkit-input-placeholder {\n    color: ", ";\n    font-family: ", ";\n    font-size: ", ";\n    font-weight: ", ";\n    letter-spacing: ", ";\n    line-height: ", ";\n    text-align: ", ";\n    text-transform: ", ";\n  }\n\n  ", ";\n"], ["\n  background-color: ",
    ";\n  border: ",
    ";\n  border-radius: ",
    ";\n  color: ",
    ";\n  display: inline-block;\n  font-family: ",
    ";\n  font-size: ",
    ";\n  font-stretch: normal;\n  font-style: normal;\n  font-weight: ",
    ";\n  letter-spacing: ",
    ";\n  line-height: ",
    ";\n  margin: 0;\n  overflow: auto;\n  padding: 1em;\n  resize: none;\n  width: 100%;\n\n  &:focus {\n    outline: none;\n    border: ",
    ";\n  }\n\n  &::-webkit-input-placeholder {\n    color: ",
    ";\n    font-family: ", ";\n    font-size: ", ";\n    font-weight: ", ";\n    letter-spacing: ", ";\n    line-height: ", ";\n    text-align: ", ";\n    text-transform: ", ";\n  }\n\n  ",
    ";\n"])), function (_a) {
    var backgroundColor = _a.backgroundColor, theme = _a.theme;
    return backgroundColor || theme.colors.white;
}, function (_a) {
    var border = _a.border, hasError = _a.hasError, theme = _a.theme;
    return hasError ? "0.5px solid " + theme.colors.cherryRed : border || "1px solid " + theme.colors.darkGrey;
}, function (_a) {
    var borderRadius = _a.borderRadius;
    return borderRadius || '0.5em';
}, function (_a) {
    var color = _a.color, theme = _a.theme;
    return color || theme.colors.black;
}, function (_a) {
    var fontFamily = _a.fontFamily, theme = _a.theme;
    return fontFamily || theme.typography.text.fontFamily;
}, function (_a) {
    var fontSize = _a.fontSize, theme = _a.theme;
    return fontSize || theme.typography.text.fontSize;
}, function (_a) {
    var fontWeight = _a.fontWeight, theme = _a.theme;
    return fontWeight || theme.typography.text.fontWeight;
}, function (_a) {
    var letterSpacing = _a.letterSpacing, theme = _a.theme;
    return letterSpacing || theme.typography.text.letterSpacing;
}, function (_a) {
    var lineHeight = _a.lineHeight, theme = _a.theme;
    return lineHeight || theme.typography.text.lineHeight;
}, function (_a) {
    var outlineColor = _a.outlineColor, theme = _a.theme;
    return "0.5px solid " + outlineColor || "0.5px solid " + theme.colors.skyBlue;
}, function (_a) {
    var placeholderColor = _a.placeholderColor, theme = _a.theme;
    return placeholderColor || theme.colors.lightBlack;
}, function (_a) {
    var theme = _a.theme;
    return theme.typography.text.fontFamily;
}, function (_a) {
    var theme = _a.theme;
    return theme.typography.text.fontSize;
}, function (_a) {
    var theme = _a.theme;
    return theme.typography.text.fontWeight;
}, function (_a) {
    var theme = _a.theme;
    return theme.typography.text.letterSpacing;
}, function (_a) {
    var theme = _a.theme;
    return theme.typography.text.lineHeight;
}, function (_a) {
    var theme = _a.theme;
    return theme.typography.text.textAlign;
}, function (_a) {
    var theme = _a.theme;
    return theme.typography.text.textTransform;
}, function (_a) {
    var desktop = _a.desktop, laptop = _a.laptop, laptopL = _a.laptopL, mobileL = _a.mobileL, mobileM = _a.mobileM, mobileS = _a.mobileS, tablet = _a.tablet;
    return util_functions_1.responsiveComponent({
        desktop: desktop,
        laptop: laptop,
        laptopL: laptopL,
        mobileL: mobileL,
        mobileM: mobileM,
        mobileS: mobileS,
        tablet: tablet,
    });
});
exports.TextAreaWithFieldLabel = field_label_1.withFieldLabel(exports.TextArea);
var templateObject_1;
