// @react-pakistan/react-commons-collection/audio/audio.js
"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Audio = void 0;
var util_functions_1 = require("@react-pakistan/util-functions");
var styled_components_1 = __importDefault(require("styled-components"));
exports.Audio = styled_components_1.default.audio.attrs(function (_a) {
    var autoplay = _a.autoplay, controls = _a.controls, loop = _a.loop, muted = _a.muted, preload = _a.preload, src = _a.src;
    return ({
        autoplay: autoplay || true,
        controls: controls || true,
        loop: loop || false,
        muted: muted || false,
        preload: preload || 'auto',
        src: src,
    });
})(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n  width: 100%;\n\n  &::-webkit-media-controls-panel {\n    background-color: ", ";\n  }\n\n  &::-webkit-media-controls-current-time-display {\n    color: ", ";\n  }\n\n  &::-webkit-media-controls-time-remaining-display {\n    color: ", ";\n  }\n\n  ", ";\n"], ["\n  width: 100%;\n\n  &::-webkit-media-controls-panel {\n    background-color: ", ";\n  }\n\n  &::-webkit-media-controls-current-time-display {\n    color: ", ";\n  }\n\n  &::-webkit-media-controls-time-remaining-display {\n    color: ", ";\n  }\n\n  ",
    ";\n"])), function (_a) {
    var controlPanelColor = _a.controlPanelColor;
    return controlPanelColor || '';
}, function (_a) {
    var currentTimeColor = _a.currentTimeColor;
    return currentTimeColor || '';
}, function (_a) {
    var remainingTimeColor = _a.remainingTimeColor;
    return remainingTimeColor || '';
}, function (_a) {
    var desktop = _a.desktop, laptop = _a.laptop, laptopL = _a.laptopL, mobileL = _a.mobileL, mobileM = _a.mobileM, mobileS = _a.mobileS, tablet = _a.tablet;
    return util_functions_1.responsiveComponent({
        desktop: desktop,
        laptop: laptop,
        laptopL: laptopL,
        mobileL: mobileL,
        mobileM: mobileM,
        mobileS: mobileS,
        tablet: tablet,
    });
});
var templateObject_1;
