// @react-pakistan/react-commons-collection/grid/grid.js
"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Grid = void 0;
var util_functions_1 = require("@react-pakistan/util-functions");
var styled_components_1 = __importDefault(require("styled-components"));
exports.Grid = styled_components_1.default.div(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n\talign-items: ", ";\n\tdisplay: grid;\n\tgrid-column-gap: ", ";\n\tgrid-row-gap: ", ";\n\tgrid-template-columns: ", ";\n\tgrid-template-rows: ", ";\n  height: ", ";\n\tjustify-items: ", ";\n\twidth: ", ";\n\n  ", ";\n"], ["\n\talign-items: ", ";\n\tdisplay: grid;\n\tgrid-column-gap: ", ";\n\tgrid-row-gap: ", ";\n\tgrid-template-columns: ", ";\n\tgrid-template-rows: ", ";\n  height: ", ";\n\tjustify-items: ", ";\n\twidth: ", ";\n\n  ",
    ";\n"])), function (_a) {
    var alignItems = _a.alignItems;
    return alignItems || 'center';
}, function (_a) {
    var columnGap = _a.columnGap;
    return columnGap || '0';
}, function (_a) {
    var rowGap = _a.rowGap;
    return rowGap || '0';
}, function (_a) {
    var gridColumn = _a.gridColumn;
    return gridColumn || '1fr 1fr';
}, function (_a) {
    var gridRow = _a.gridRow;
    return gridRow || '1fr';
}, function (_a) {
    var height = _a.height;
    return height || 'auto';
}, function (_a) {
    var justifyItems = _a.justifyItems;
    return justifyItems || 'stretch';
}, function (_a) {
    var width = _a.width;
    return width || '100%';
}, function (_a) {
    var desktop = _a.desktop, laptop = _a.laptop, laptopL = _a.laptopL, mobileL = _a.mobileL, mobileM = _a.mobileM, mobileS = _a.mobileS, tablet = _a.tablet;
    return util_functions_1.responsiveComponent({
        desktop: desktop,
        laptop: laptop,
        laptopL: laptopL,
        mobileL: mobileL,
        mobileM: mobileM,
        mobileS: mobileS,
        tablet: tablet,
    });
});
var templateObject_1;
