// @react-pakistan/react-commons-collection/video/video.js
"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Video = void 0;
var util_functions_1 = require("@react-pakistan/util-functions");
var styled_components_1 = __importDefault(require("styled-components"));
exports.Video = styled_components_1.default.video.attrs(function (_a) {
    var autoplay = _a.autoplay, controls = _a.controls, loop = _a.loop, muted = _a.muted, poster = _a.poster, preload = _a.preload, src = _a.src, width = _a.width;
    return ({
        autoplay: autoplay || true,
        controls: controls || true,
        loop: loop || false,
        muted: muted || false,
        poster: poster || '',
        preload: preload || 'auto',
        src: src,
        width: width || '100%',
    });
})(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n\n  ", ";\n"], ["\n\n  ",
    ";\n"])), function (_a) {
    var desktop = _a.desktop, laptop = _a.laptop, laptopL = _a.laptopL, mobileL = _a.mobileL, mobileM = _a.mobileM, mobileS = _a.mobileS, tablet = _a.tablet;
    return util_functions_1.responsiveComponent({
        desktop: desktop,
        laptop: laptop,
        laptopL: laptopL,
        mobileL: mobileL,
        mobileM: mobileM,
        mobileS: mobileS,
        tablet: tablet,
    });
});
var templateObject_1;
