// @react-pakistan/react-commons-collection/a/a.js
"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.A = void 0;
var util_functions_1 = require("@react-pakistan/util-functions");
var styled_components_1 = __importDefault(require("styled-components"));
exports.A = styled_components_1.default.a.attrs(function (_a) {
    var download = _a.download, href = _a.href, hreflang = _a.hreflang, media = _a.media, referrerpolicy = _a.referrerpolicy, rel = _a.rel, target = _a.target, type = _a.type;
    return ({
        download: download || null,
        href: href,
        hreflang: hreflang || '',
        media: media || '',
        referrerpolicy: referrerpolicy || '',
        rel: rel || '',
        target: target,
        type: type || '',
    });
})(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n  color: ", ";\n  text-decoration: none;\n\n  ", ";\n"], ["\n  color: ",
    ";\n  text-decoration: none;\n\n  ",
    ";\n"])), function (_a) {
    var color = _a.color, theme = _a.theme;
    return color || theme.colors.darkGrey;
}, function (_a) {
    var desktop = _a.desktop, laptop = _a.laptop, laptopL = _a.laptopL, mobileL = _a.mobileL, mobileM = _a.mobileM, mobileS = _a.mobileS, tablet = _a.tablet;
    return util_functions_1.responsiveComponent({
        desktop: desktop,
        laptop: laptop,
        laptopL: laptopL,
        mobileL: mobileL,
        mobileM: mobileM,
        mobileS: mobileS,
        tablet: tablet,
    });
});
var templateObject_1;
