// @react-pakistan/react-commons-collection/h1/h1.js
"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.H1 = void 0;
var util_functions_1 = require("@react-pakistan/util-functions");
var styled_components_1 = __importDefault(require("styled-components"));
exports.H1 = styled_components_1.default.h1(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n  color: ", ";\n  font-family: ", ";\n  font-size: ", ";\n  font-stretch: normal;\n  font-style: normal;\n  font-weight: ", ";\n  letter-spacing: ", ";\n  line-height: ", ";\n  margin: 0;\n  padding: 0;\n  text-align: ", ";\n  text-transform: ", ";\n\n  ", ";\n"], ["\n  color: ",
    ";\n  font-family: ",
    ";\n  font-size: ",
    ";\n  font-stretch: normal;\n  font-style: normal;\n  font-weight: ",
    ";\n  letter-spacing: ",
    ";\n  line-height: ",
    ";\n  margin: 0;\n  padding: 0;\n  text-align: ",
    ";\n  text-transform: ",
    ";\n\n  ",
    ";\n"])), function (_a) {
    var color = _a.color, theme = _a.theme;
    return color || theme.colors.black;
}, function (_a) {
    var fontFamily = _a.fontFamily, theme = _a.theme;
    return fontFamily || theme.typography.h1.fontFamily;
}, function (_a) {
    var fontSize = _a.fontSize, theme = _a.theme;
    return fontSize || theme.typography.h1.fontSize;
}, function (_a) {
    var fontWeight = _a.fontWeight, theme = _a.theme;
    return fontWeight || theme.typography.h1.fontWeight;
}, function (_a) {
    var letterSpacing = _a.letterSpacing, theme = _a.theme;
    return letterSpacing || theme.typography.h1.letterSpacing;
}, function (_a) {
    var lineHeight = _a.lineHeight, theme = _a.theme;
    return lineHeight || theme.typography.h1.lineHeight;
}, function (_a) {
    var textAlign = _a.textAlign, theme = _a.theme;
    return textAlign || theme.typography.h1.textAlign;
}, function (_a) {
    var textTransform = _a.textTransform, theme = _a.theme;
    return textTransform || theme.typography.h1.textTransform;
}, function (_a) {
    var desktop = _a.desktop, laptop = _a.laptop, laptopL = _a.laptopL, mobileL = _a.mobileL, mobileM = _a.mobileM, mobileS = _a.mobileS, tablet = _a.tablet;
    return util_functions_1.responsiveComponent({
        desktop: desktop,
        laptop: laptop,
        laptopL: laptopL,
        mobileL: mobileL,
        mobileM: mobileM,
        mobileS: mobileS,
        tablet: tablet,
    });
});
var templateObject_1;
