// @react-pakistan/react-commons-collection/button/button.js
"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Button = void 0;
var util_functions_1 = require("@react-pakistan/util-functions");
var styled_components_1 = __importStar(require("styled-components"));
var type_1 = require("./type");
exports.Button = styled_components_1.default.button.attrs(function (_a) {
    var autoFocus = _a.autoFocus, disabled = _a.disabled, form = _a.form, formaction = _a.formaction, formenctype = _a.formenctype, formmethod = _a.formmethod, formnovalidate = _a.formnovalidate, formtarget = _a.formtarget, name = _a.name, type = _a.type, value = _a.value;
    return ({
        autoFocus: autoFocus || false,
        disabled: disabled || false,
        form: form || '',
        formaction: formaction || '',
        formenctype: formenctype || type_1.BUTTON_FORM_ENCTYPE.TEXT,
        formmethod: formmethod || type_1.BUTTON_FORM_METHODS.GET,
        formnovalidate: formnovalidate || false,
        formtarget: formtarget || type_1.BUTTON_FORM_TARGET.SELF,
        name: name || '',
        type: type || type_1.BUTTON_TYPE.BUTTON,
        value: value || '',
    });
})(templateObject_2 || (templateObject_2 = __makeTemplateObject(["\n  background-color: ", ";\n  border-radius: ", ";\n  border: ", ";\n  height: ", ";\n  padding: ", ";\n  pointer-events: ", ";\n  transition: all 0.3s ease-in-out;\n  width: ", ";\n\n  &:hover, &:focus, &:active {\n    cursor: ", "\n  }\n\n  &:focus {\n    outline: none;\n  }\n\n  @media ", " {\n    ", ";\n  }\n\n  ", ";\n"], ["\n  background-color: ",
    ";\n  border-radius: ", ";\n  border: ", ";\n  height: ", ";\n  padding: ", ";\n  pointer-events: ", ";\n  transition: all 0.3s ease-in-out;\n  width: ", ";\n\n  &:hover, &:focus, &:active {\n    cursor: ", "\n  }\n\n  &:focus {\n    outline: none;\n  }\n\n  @media ", " {\n    ",
    ";\n  }\n\n  ",
    ";\n"])), function (_a) {
    var backgroundColor = _a.backgroundColor, theme = _a.theme;
    return (backgroundColor || theme.colors.white);
}, function (_a) {
    var borderRadius = _a.borderRadius;
    return borderRadius || '';
}, function (_a) {
    var border = _a.border;
    return border || 'none';
}, function (_a) {
    var height = _a.height;
    return height || 'auto';
}, function (_a) {
    var padding = _a.padding;
    return padding || '';
}, function (_a) {
    var disabled = _a.disabled;
    return disabled ? 'none' : 'auto';
}, function (_a) {
    var width = _a.width;
    return width || 'auto';
}, function (_a) {
    var disabled = _a.disabled;
    return disabled ? 'not-allowed' : 'pointer';
}, function (_a) {
    var theme = _a.theme;
    return theme.breakpoints.mobileS;
}, function (_a) {
    var mobileS = _a.mobileS;
    return mobileS && styled_components_1.css(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n      ", "\n    "], ["\n      ", "\n    "])), mobileS);
}, function (_a) {
    var desktop = _a.desktop, laptop = _a.laptop, laptopL = _a.laptopL, mobileL = _a.mobileL, mobileM = _a.mobileM, mobileS = _a.mobileS, tablet = _a.tablet;
    return util_functions_1.responsiveComponent({
        desktop: desktop,
        laptop: laptop,
        laptopL: laptopL,
        mobileL: mobileL,
        mobileM: mobileM,
        mobileS: mobileS,
        tablet: tablet,
    });
});
var templateObject_1, templateObject_2;
