// @react-pakistan/react-commons-collection/button/type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BUTTON_TYPE = exports.BUTTON_FORM_TARGET = exports.BUTTON_FORM_METHODS = exports.BUTTON_FORM_ENCTYPE = void 0;
var BUTTON_FORM_ENCTYPE;
(function (BUTTON_FORM_ENCTYPE) {
    BUTTON_FORM_ENCTYPE["APPLICATION"] = "application/x-www-form-urlencoded";
    BUTTON_FORM_ENCTYPE["MULTIPART"] = "multipart/form-data";
    BUTTON_FORM_ENCTYPE["TEXT"] = "text/plain";
})(BUTTON_FORM_ENCTYPE = exports.BUTTON_FORM_ENCTYPE || (exports.BUTTON_FORM_ENCTYPE = {}));
var BUTTON_FORM_METHODS;
(function (BUTTON_FORM_METHODS) {
    BUTTON_FORM_METHODS["GET"] = "get";
    BUTTON_FORM_METHODS["POST"] = "post";
})(BUTTON_FORM_METHODS = exports.BUTTON_FORM_METHODS || (exports.BUTTON_FORM_METHODS = {}));
var BUTTON_FORM_TARGET;
(function (BUTTON_FORM_TARGET) {
    BUTTON_FORM_TARGET["BLANK"] = "_blank";
    BUTTON_FORM_TARGET["PARENT"] = "_parent";
    BUTTON_FORM_TARGET["SELF"] = "_self";
    BUTTON_FORM_TARGET["TOP"] = "_top";
})(BUTTON_FORM_TARGET = exports.BUTTON_FORM_TARGET || (exports.BUTTON_FORM_TARGET = {}));
var BUTTON_TYPE;
(function (BUTTON_TYPE) {
    BUTTON_TYPE["BUTTON"] = "button";
    BUTTON_TYPE["RESET"] = "reset";
    BUTTON_TYPE["SUBMIT"] = "submit";
})(BUTTON_TYPE = exports.BUTTON_TYPE || (exports.BUTTON_TYPE = {}));
