// @react-pakistan/react-commons-collection/styled-app.js
"use strict";
// tslint:disable typedef
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.StyledStory = exports.StyledApp = exports.ThemeProvider = exports.keyframes = exports.createGlobalStyle = exports.css = exports.styled = void 0;
var misc_color_1 = require("@react-pakistan/react-logo-collection/misc-color");
var util_functions_1 = require("@react-pakistan/util-functions");
var React = __importStar(require("react"));
var styledComponents = __importStar(require("styled-components"));
// https://www.styled-components.com/docs/api#define-a-theme-interface
var _a = styledComponents, styled = _a.default, css = _a.css, createGlobalStyle = _a.createGlobalStyle, keyframes = _a.keyframes, ThemeProvider = _a.ThemeProvider;
exports.styled = styled;
exports.css = css;
exports.createGlobalStyle = createGlobalStyle;
exports.keyframes = keyframes;
exports.ThemeProvider = ThemeProvider;
exports.StyledApp = styled.div(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n  display: inline-block;\n  font-family: ", ";\n  width: 100%;\n  *,\n  *::before,\n  *::after {\n    box-sizing: border-box;\n  }\n"], ["\n  display: inline-block;\n  font-family: ", ";\n  width: 100%;\n  *,\n  *::before,\n  *::after {\n    box-sizing: border-box;\n  }\n"])), function (props) { return props.theme.typography.fontFamily; });
exports.StyledStory = function (_a) {
    var children = _a.children;
    return (React.createElement(ThemeProvider, { theme: util_functions_1.theme },
        React.createElement(exports.StyledApp, null, children),
        React.createElement(misc_color_1.MiscColor1, { fontSize: '200px' })));
};
var templateObject_1;
