// @react-pakistan/react-commons-collection/column-container/column-container.js
"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ColumnContainer = void 0;
var util_functions_1 = require("@react-pakistan/util-functions");
var styled_components_1 = __importDefault(require("styled-components"));
exports.ColumnContainer = styled_components_1.default.div(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n  align-items: ", ";\n  display: flex;\n  flex-direction: column;\n  justify-content: ", ";\n\n  ", ";\n"], ["\n  align-items: ",
    ";\n  display: flex;\n  flex-direction: column;\n  justify-content: ",
    ";\n\n  ",
    ";\n"])), function (_a) {
    var alignItems = _a.alignItems;
    return alignItems || 'center';
}, function (_a) {
    var justifyContent = _a.justifyContent;
    return justifyContent || 'center';
}, function (_a) {
    var desktop = _a.desktop, laptop = _a.laptop, laptopL = _a.laptopL, mobileL = _a.mobileL, mobileM = _a.mobileM, mobileS = _a.mobileS, tablet = _a.tablet;
    return util_functions_1.responsiveComponent({
        desktop: desktop,
        laptop: laptop,
        laptopL: laptopL,
        mobileL: mobileL,
        mobileM: mobileM,
        mobileS: mobileS,
        tablet: tablet,
    });
});
var templateObject_1;
