// @react-pakistan/react-commons-collection/progress-bar/progress-bar.js
"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProgressBar = void 0;
var util_functions_1 = require("@react-pakistan/util-functions");
var styled_components_1 = __importDefault(require("styled-components"));
exports.ProgressBar = styled_components_1.default.progress.attrs(function (_a) {
    var max = _a.max, min = _a.min, value = _a.value;
    return ({
        max: max,
        min: min,
        value: value,
    });
})(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n  border: 0px none;\n  background-color: ", ";\n  height: ", ";\n  width: ", ";\n\n  &::-webkit-progress-bar {\n    background-color: ", ";\n  }\n\n  &::-webkit-progress-value {\n    background-color: ", ";\n    transition: ", ";\n  }\n\n  &::-moz-progress-bar {\n    background-color: ", ";\n  }\n\n  ", ";\n"], ["\n  border: 0px none;\n  background-color: ",
    ";\n  height: ", ";\n  width: ", ";\n\n  &::-webkit-progress-bar {\n    background-color: ",
    ";\n  }\n\n  &::-webkit-progress-value {\n    background-color: ",
    ";\n    transition: ", ";\n  }\n\n  &::-moz-progress-bar {\n    background-color: ",
    ";\n  }\n\n  ",
    ";\n"])), function (_a) {
    var backgroundColorBar = _a.backgroundColorBar, theme = _a.theme;
    return backgroundColorBar || theme.colors.cherryRed;
}, function (_a) {
    var height = _a.height;
    return height || '0.25em';
}, function (_a) {
    var width = _a.width;
    return width || '100%';
}, function (_a) {
    var backgroundColorBar = _a.backgroundColorBar, theme = _a.theme;
    return backgroundColorBar || theme.colors.cherryRed;
}, function (_a) {
    var backgroundColorValue = _a.backgroundColorValue, theme = _a.theme;
    return backgroundColorValue || theme.colors.lightBlack;
}, function (_a) {
    var transition = _a.transition;
    return transition || '2.5s ease-out';
}, function (_a) {
    var backgroundColorBar = _a.backgroundColorBar, theme = _a.theme;
    return backgroundColorBar || theme.colors.cherryRed;
}, function (_a) {
    var desktop = _a.desktop, laptop = _a.laptop, laptopL = _a.laptopL, mobileL = _a.mobileL, mobileM = _a.mobileM, mobileS = _a.mobileS, tablet = _a.tablet;
    return util_functions_1.responsiveComponent({
        desktop: desktop,
        laptop: laptop,
        laptopL: laptopL,
        mobileL: mobileL,
        mobileM: mobileM,
        mobileS: mobileS,
        tablet: tablet,
    });
});
var templateObject_1;
