// @react-pakistan/react-commons-collection/check-box/props.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.checkBoxWithLabelProps = exports.checkBoxProps = void 0;
var misc_grey_1 = require("@react-pakistan/react-icon-collection/misc-grey");
var util_functions_1 = require("@react-pakistan/util-functions");
var react_1 = __importDefault(require("react"));
exports.checkBoxProps = {
    check: react_1.default.createElement(misc_grey_1.MiscGrey20, { fill: util_functions_1.theme.colors.white, fontSize: '20px' }),
    checkedColor: util_functions_1.theme.colors.skyBlue,
    desktop: util_functions_1.desktop,
    laptop: util_functions_1.laptop,
    laptopL: util_functions_1.laptopL,
    mobileL: util_functions_1.mobileL,
    mobileM: util_functions_1.mobileM,
    mobileS: util_functions_1.mobileS,
    tablet: util_functions_1.tablet,
    unCheckedColor: util_functions_1.theme.colors.lighterBlack,
};
exports.checkBoxWithLabelProps = {
    check: react_1.default.createElement(misc_grey_1.MiscGrey20, { fill: util_functions_1.theme.colors.white, fontSize: '20px' }),
    checkedColor: util_functions_1.theme.colors.skyBlue,
    desktop: util_functions_1.desktop,
    hasError: false,
    helperText: 'There is an error',
    helperTextColor: util_functions_1.theme.colors.cherryRed,
    label: 'Pick one',
    laptop: util_functions_1.laptop,
    laptopL: util_functions_1.laptopL,
    mobileL: util_functions_1.mobileL,
    mobileM: util_functions_1.mobileM,
    mobileS: util_functions_1.mobileS,
    tablet: util_functions_1.tablet,
    unCheckedColor: util_functions_1.theme.colors.lighterBlack,
};
