// @react-pakistan/react-commons-collection/check-box/styled.js
"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.StyledCheckbox = exports.Check = exports.HideNativeCheckbox = exports.CheckBoxContainer = void 0;
var util_functions_1 = require("@react-pakistan/util-functions");
var styled_components_1 = __importDefault(require("styled-components"));
exports.CheckBoxContainer = styled_components_1.default.div(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n  display: inline-block;\n  height: ", ";\n\n  ", ";\n"], ["\n  display: inline-block;\n  height: ", ";\n\n  ",
    ";\n"])), function (_a) {
    var theme = _a.theme;
    return theme.spacing.default;
}, function (_a) {
    var desktop = _a.desktop, laptop = _a.laptop, laptopL = _a.laptopL, mobileL = _a.mobileL, mobileM = _a.mobileM, mobileS = _a.mobileS, tablet = _a.tablet;
    return util_functions_1.responsiveComponent({
        desktop: desktop,
        laptop: laptop,
        laptopL: laptopL,
        mobileL: mobileL,
        mobileM: mobileM,
        mobileS: mobileS,
        tablet: tablet,
    });
});
exports.HideNativeCheckbox = styled_components_1.default.input.attrs(function () { return ({
    type: 'checkbox',
}); })(templateObject_2 || (templateObject_2 = __makeTemplateObject(["\n  border: 0;\n  clip: rect(0 0 0 0);\n  height: 0px;\n  margin: 0px;\n  overflow: hidden;\n  padding: 0;\n  position: absolute;\n  white-space: nowrap;\n  width: 0px;\n"], ["\n  border: 0;\n  clip: rect(0 0 0 0);\n  height: 0px;\n  margin: 0px;\n  overflow: hidden;\n  padding: 0;\n  position: absolute;\n  white-space: nowrap;\n  width: 0px;\n"])));
exports.Check = styled_components_1.default.svg(templateObject_3 || (templateObject_3 = __makeTemplateObject(["\n  fill: none;\n  stroke-width: 2px;\n  stroke: ", ";\n"], ["\n  fill: none;\n  stroke-width: 2px;\n  stroke: ", ";\n"])), function (_a) {
    var theme = _a.theme;
    return theme.colors.white;
});
exports.StyledCheckbox = styled_components_1.default.div(templateObject_4 || (templateObject_4 = __makeTemplateObject(["\n  background-color: ", ";\n  border-radius: ", ";\n  display: inline-block;\n  height: ", ";\n  transition: all 150ms;\n  width: ", ";\n\n  ", ":focus + & {\n    box-shadow: ", ";\n  }\n\n  ", " {\n    visibility: ", "\n  }\n"], ["\n  background-color: ",
    ";\n  border-radius: ", ";\n  display: inline-block;\n  height: ", ";\n  transition: all 150ms;\n  width: ", ";\n\n  ", ":focus + & {\n    box-shadow: ", ";\n  }\n\n  ", " {\n    visibility: ",
    "\n  }\n"])), function (_a) {
    var checkedColor = _a.checkedColor, isChecked = _a.isChecked, theme = _a.theme, unCheckedColor = _a.unCheckedColor;
    return (isChecked ? (checkedColor || theme.colors.skyBlue) : (unCheckedColor || theme.colors.lighterBlack));
}, function (_a) {
    var theme = _a.theme;
    return theme.radius.default;
}, function (_a) {
    var theme = _a.theme;
    return theme.spacing.default;
}, function (_a) {
    var theme = _a.theme;
    return theme.spacing.default;
}, exports.HideNativeCheckbox, function (_a) {
    var theme = _a.theme;
    return theme.shadow.default;
}, exports.Check, function (_a) {
    var isChecked = _a.isChecked;
    return isChecked ? 'visible' : 'hidden';
});
var templateObject_1, templateObject_2, templateObject_3, templateObject_4;
