// @react-pakistan/util-functions/constants/social-media-sizes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MOBILE_PLATFORM_MEDIA_SIZES = exports.MOBILE_PLATFORM_TYPES = exports.SOCIAL_MEDIA_POST_SIZES = exports.SOCIAL_MEDIA_POST_TYPES = void 0;
var SOCIAL_MEDIA_POST_TYPES;
(function (SOCIAL_MEDIA_POST_TYPES) {
    SOCIAL_MEDIA_POST_TYPES["FACEBOOK_COVER_PHOTO"] = "FACEBOOK_COVER_PHOTO";
    SOCIAL_MEDIA_POST_TYPES["FACEBOOK_EVENT_COVER"] = "FACEBOOK_EVENT_COVER";
    SOCIAL_MEDIA_POST_TYPES["FACEBOOK_PROFILE_PHOTO"] = "FACEBOOK_PROFILE_PHOTO";
    SOCIAL_MEDIA_POST_TYPES["FACEBOOK_POST_SHARE"] = "FACEBOOK_POST_SHARE";
    SOCIAL_MEDIA_POST_TYPES["FACEBOOK_STORIES"] = "FACEBOOK_STORIES";
    SOCIAL_MEDIA_POST_TYPES["INSTAGRAM_PROFILE_PHOTO"] = "INSTAGRAM_PROFILE_PHOTO";
    SOCIAL_MEDIA_POST_TYPES["INSTAGRAM_PHOTO_SHARE"] = "INSTAGRAM_PHOTO_SHARE";
    SOCIAL_MEDIA_POST_TYPES["INSTAGRAM_STORIES"] = "INSTAGRAM_STORIES";
    SOCIAL_MEDIA_POST_TYPES["LINKEDIN_PROFILE_PHOTO"] = "LINKEDIN_PROFILE_PHOTO";
    SOCIAL_MEDIA_POST_TYPES["LINKEDIN_PROFILE_BACKGROUND"] = "LINKEDIN_PROFILE_BACKGROUND";
    SOCIAL_MEDIA_POST_TYPES["LINKEDIN_COMPANY_LOGO"] = "LINKEDIN_COMPANY_LOGO";
    SOCIAL_MEDIA_POST_TYPES["LINKEDIN_COMPANY_COVER"] = "LINKEDIN_COMPANY_COVER";
    SOCIAL_MEDIA_POST_TYPES["LINKEDIN_SHARED_IMAGE"] = "LINKEDIN_SHARED_IMAGE";
    SOCIAL_MEDIA_POST_TYPES["LINKEDIN_SHARED_IMAGE_LINK"] = "LINKEDIN_SHARED_IMAGE_LINK";
    SOCIAL_MEDIA_POST_TYPES["PINTEREST_PROFILE_PHOTO"] = "PINTEREST_PROFILE_PHOTO";
    SOCIAL_MEDIA_POST_TYPES["PINTEREST_BOARD_COVER"] = "PINTEREST_BOARD_COVER";
    SOCIAL_MEDIA_POST_TYPES["PINTEREST_SHARED_PIN"] = "PINTEREST_SHARED_PIN";
    SOCIAL_MEDIA_POST_TYPES["PINTEREST_PROFILE_COVER"] = "PINTEREST_PROFILE_COVER";
    SOCIAL_MEDIA_POST_TYPES["SNAPCHAT_SHARED_IMAGE"] = "SNAPCHAT_SHARED_IMAGE";
    SOCIAL_MEDIA_POST_TYPES["TWITTER_HEADER"] = "TWITTER_HEADER";
    SOCIAL_MEDIA_POST_TYPES["TWITTER_PROFILE_PHOTO"] = "TWITTER_PROFILE_PHOTO";
    SOCIAL_MEDIA_POST_TYPES["TWITTER_POST_SHARE"] = "TWITTER_POST_SHARE";
    SOCIAL_MEDIA_POST_TYPES["YOUTUBE_CHANNEL_PROFILE"] = "YOUTUBE_CHANNEL_PROFILE";
    SOCIAL_MEDIA_POST_TYPES["YOUTUBE_CHANNEL_COVER"] = "YOUTUBE_CHANNEL_COVER";
})(SOCIAL_MEDIA_POST_TYPES = exports.SOCIAL_MEDIA_POST_TYPES || (exports.SOCIAL_MEDIA_POST_TYPES = {}));
exports.SOCIAL_MEDIA_POST_SIZES = {
    [SOCIAL_MEDIA_POST_TYPES.FACEBOOK_COVER_PHOTO]: {
        height: '360px',
        width: '820px',
    },
    [SOCIAL_MEDIA_POST_TYPES.FACEBOOK_EVENT_COVER]: {
        height: '1080px',
        width: '1920px',
    },
    [SOCIAL_MEDIA_POST_TYPES.FACEBOOK_PROFILE_PHOTO]: {
        height: '400px',
        width: '400px',
    },
    [SOCIAL_MEDIA_POST_TYPES.FACEBOOK_POST_SHARE]: {
        height: '630px',
        width: '1200px',
    },
    [SOCIAL_MEDIA_POST_TYPES.FACEBOOK_STORIES]: {
        height: '1920px',
        width: '1080px',
    },
    [SOCIAL_MEDIA_POST_TYPES.INSTAGRAM_PROFILE_PHOTO]: {
        height: '400px',
        width: '400px',
    },
    [SOCIAL_MEDIA_POST_TYPES.INSTAGRAM_PHOTO_SHARE]: {
        height: '1080px',
        width: '1080px',
    },
    [SOCIAL_MEDIA_POST_TYPES.INSTAGRAM_STORIES]: {
        height: '1920px',
        width: '1080px',
    },
    [SOCIAL_MEDIA_POST_TYPES.LINKEDIN_PROFILE_PHOTO]: {
        height: '400px',
        width: '400px',
    },
    [SOCIAL_MEDIA_POST_TYPES.LINKEDIN_PROFILE_BACKGROUND]: {
        height: '396px',
        width: '1584px',
    },
    [SOCIAL_MEDIA_POST_TYPES.LINKEDIN_COMPANY_LOGO]: {
        height: '300px',
        width: '300px',
    },
    [SOCIAL_MEDIA_POST_TYPES.LINKEDIN_COMPANY_COVER]: {
        height: '768px',
        width: '1536px',
    },
    [SOCIAL_MEDIA_POST_TYPES.LINKEDIN_SHARED_IMAGE]: {
        height: '736px',
        width: '1200px',
    },
    [SOCIAL_MEDIA_POST_TYPES.LINKEDIN_SHARED_IMAGE_LINK]: {
        height: '628px',
        width: '1200px',
    },
    [SOCIAL_MEDIA_POST_TYPES.PINTEREST_PROFILE_PHOTO]: {
        height: '400px',
        width: '400px',
    },
    [SOCIAL_MEDIA_POST_TYPES.PINTEREST_BOARD_COVER]: {
        height: '600px',
        width: '600px',
    },
    [SOCIAL_MEDIA_POST_TYPES.PINTEREST_SHARED_PIN]: {
        height: '900px',
        width: '600px',
    },
    [SOCIAL_MEDIA_POST_TYPES.PINTEREST_PROFILE_COVER]: {
        height: '1080px',
        width: '1920px',
    },
    [SOCIAL_MEDIA_POST_TYPES.SNAPCHAT_SHARED_IMAGE]: {
        height: '1920px',
        width: '1080px',
    },
    [SOCIAL_MEDIA_POST_TYPES.TWITTER_HEADER]: {
        height: '500px',
        width: '1500px',
    },
    [SOCIAL_MEDIA_POST_TYPES.TWITTER_PROFILE_PHOTO]: {
        height: '400px',
        width: '400px',
    },
    [SOCIAL_MEDIA_POST_TYPES.TWITTER_POST_SHARE]: {
        height: '630px',
        width: '1200px',
    },
    [SOCIAL_MEDIA_POST_TYPES.YOUTUBE_CHANNEL_PROFILE]: {
        height: '800px',
        width: '800px',
    },
    [SOCIAL_MEDIA_POST_TYPES.YOUTUBE_CHANNEL_COVER]: {
        height: '423px',
        width: '1546px',
    },
};
var MOBILE_PLATFORM_TYPES;
(function (MOBILE_PLATFORM_TYPES) {
    MOBILE_PLATFORM_TYPES["ANDROID"] = "android";
    MOBILE_PLATFORM_TYPES["IOS"] = "ios";
    MOBILE_PLATFORM_TYPES["WINDOWS"] = "windows";
})(MOBILE_PLATFORM_TYPES = exports.MOBILE_PLATFORM_TYPES || (exports.MOBILE_PLATFORM_TYPES = {}));
exports.MOBILE_PLATFORM_MEDIA_SIZES = {
    [MOBILE_PLATFORM_TYPES.ANDROID]: {
        appIcons: {
            googlePlay: {
                height: '512px',
                width: '512px',
            },
            xxhdpi: {
                height: '144px',
                width: '144px',
            },
            xhdpi: {
                height: '96px',
                width: '96px',
            },
            hdpi: {
                height: '72px',
                width: '72px',
            },
            mdpi: {
                height: '48px',
                width: '48px',
            },
        },
        actionBarIcons: {
            xxhdpi: {
                height: '96px',
                width: '96px',
            },
            xhdpi: {
                height: '64px',
                width: '64px',
            },
            hdpi: {
                height: '48px',
                width: '48px',
            },
            mdpi: {
                height: '32px',
                width: '32px',
            },
        },
        smallContextualIcons: {
            xxhdpi: {
                height: '48px',
                width: '48px',
            },
            xhdpi: {
                height: '32px',
                width: '32px',
            },
            hdpi: {
                height: '24px',
                width: '24px',
            },
            mdpi: {
                height: '16px',
                width: '16px',
            },
        },
        notificationIcons: {
            xxhdpi: {
                height: '72px',
                width: '72px',
            },
            xhdpi: {
                height: '48px',
                width: '48px',
            },
            hdpi: {
                height: '36px',
                width: '36px',
            },
            mdpi: {
                height: '24px',
                width: '24px',
            },
        },
    },
    [MOBILE_PLATFORM_TYPES.IOS]: {
        appIcons: {
            xxxxxlarge: {
                height: '1024px',
                width: '1024px',
            },
            xxxxlarge: {
                height: '512px',
                width: '512px',
            },
            xxxlarge: {
                height: '228px',
                width: '228px',
            },
            xxlarge: {
                height: '180px',
                width: '180px',
            },
            xlarge: {
                height: '152px',
                width: '152px',
            },
            large: {
                height: '120px',
                width: '120px',
            },
            medium: {
                height: '87px',
                width: '87px',
            },
            small: {
                height: '80px',
                width: '80px',
            },
            xsmall: {
                height: '76px',
                width: '76px',
            },
            xxsmall: {
                height: '58px',
                width: '58px',
            },
            xxxsmall: {
                height: '40px',
                width: '40px',
            },
            xxxxsmall: {
                height: '29px',
                width: '29px',
            },
        },
        toolBarNavBarIcons: {
            large: {
                height: '44px',
                width: '44px',
            },
            small: {
                height: '22px',
                width: '22px',
            },
        },
        tabBarIcons: {
            large: {
                height: '50px',
                width: '50px',
            },
            small: {
                height: '25px',
                width: '25px',
            },
        },
        appleWatchNotificationCenterIcons: {
            large: {
                height: '36px',
                width: '36px',
            },
            small: {
                height: '29px',
                width: '29px',
            },
        },
        appleWatchLongLookNotificationIcons: {
            largeWatch: {
                height: '196px',
                width: '196px',
            },
            smallWatch: {
                height: '172px',
                width: '172px',
            },
        },
    },
    appleWatchMenuIcons: {
        largeWatch: {
            height: '80px',
            width: '80px',
        },
        smallWatch: {
            height: '70px',
            width: '70px',
        },
    },
    appleWatchCompanionIcons: {
        large: {
            height: '87px',
            width: '87px',
        },
        small: {
            height: '58px',
            width: '58px',
        },
    },
};
