// @react-pakistan/react-commons-collection/text-input/text-input.js
"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TextInputWithFieldLabel = exports.TextInput = void 0;
var util_functions_1 = require("@react-pakistan/util-functions");
var styled_components_1 = __importDefault(require("styled-components"));
var field_label_1 = require("../hoc/field-label");
exports.TextInput = styled_components_1.default.input.attrs(function (_a) {
    var placeholder = _a.placeholder, type = _a.type;
    return ({
        placeholder: placeholder,
        type: type,
    });
})(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n  background-color: ", ";\n  border: ", ";\n  border-radius: ", ";\n  display: inline-block;\n  height: ", ";\n  font-family: ", ";\n  font-size: ", ";\n  font-stretch: normal;\n  font-style: normal;\n  font-weight: ", ";\n  letter-spacing: ", ";\n  line-height: ", ";\n  margin: 0;\n  padding: 0 1em;\n  width: ", ";\n\n  &:focus {\n    border: ", ";\n    outline: none;\n    color: ", ";\n  }\n\n  &::-webkit-input-placeholder {\n    color: ", ";\n    font-family: ", ";\n    font-size: ", ";\n    font-weight: ", ";\n    letter-spacing: ", ";\n    line-height: ", ";\n    text-align: ", ";\n    text-transform: ", ";\n  }\n\n  ", ";\n"], ["\n  background-color: ",
    ";\n  border: ",
    ";\n  border-radius: ",
    ";\n  display: inline-block;\n  height: ", ";\n  font-family: ",
    ";\n  font-size: ",
    ";\n  font-stretch: normal;\n  font-style: normal;\n  font-weight: ",
    ";\n  letter-spacing: ",
    ";\n  line-height: ",
    ";\n  margin: 0;\n  padding: 0 1em;\n  width: ", ";\n\n  &:focus {\n    border: ",
    ";\n    outline: none;\n    color: ",
    ";\n  }\n\n  &::-webkit-input-placeholder {\n    color: ",
    ";\n    font-family: ", ";\n    font-size: ", ";\n    font-weight: ", ";\n    letter-spacing: ", ";\n    line-height: ", ";\n    text-align: ", ";\n    text-transform: ", ";\n  }\n\n  ",
    ";\n"])), function (_a) {
    var backgroundColor = _a.backgroundColor, theme = _a.theme;
    return backgroundColor || theme.colors.white;
}, function (_a) {
    var border = _a.border, hasError = _a.hasError, theme = _a.theme;
    return hasError ? "0.5px solid " + theme.colors.cherryRed : border || "1px solid " + theme.colors.darkGrey;
}, function (_a) {
    var borderRadius = _a.borderRadius;
    return borderRadius || '0.5em';
}, function (_a) {
    var height = _a.height;
    return height || '3.5em';
}, function (_a) {
    var fontFamily = _a.fontFamily, theme = _a.theme;
    return fontFamily || theme.typography.text.fontFamily;
}, function (_a) {
    var fontSize = _a.fontSize, theme = _a.theme;
    return fontSize || theme.typography.text.fontSize;
}, function (_a) {
    var fontWeight = _a.fontWeight, theme = _a.theme;
    return fontWeight || theme.typography.text.fontWeight;
}, function (_a) {
    var letterSpacing = _a.letterSpacing, theme = _a.theme;
    return letterSpacing || theme.typography.text.letterSpacing;
}, function (_a) {
    var lineHeight = _a.lineHeight, theme = _a.theme;
    return lineHeight || theme.typography.text.lineHeight;
}, function (_a) {
    var width = _a.width;
    return width || '100%';
}, function (_a) {
    var focusBorder = _a.focusBorder, theme = _a.theme;
    return "" + focusBorder || "0.5px solid " + theme.colors.skyBlue;
}, function (_a) {
    var focusColor = _a.focusColor, theme = _a.theme;
    return "" + focusColor || theme.colors.black;
}, function (_a) {
    var placeholderColor = _a.placeholderColor, theme = _a.theme;
    return placeholderColor || theme.colors.lightBlack;
}, function (_a) {
    var theme = _a.theme;
    return theme.typography.text.fontFamily;
}, function (_a) {
    var theme = _a.theme;
    return theme.typography.text.fontSize;
}, function (_a) {
    var theme = _a.theme;
    return theme.typography.text.fontWeight;
}, function (_a) {
    var theme = _a.theme;
    return theme.typography.text.letterSpacing;
}, function (_a) {
    var theme = _a.theme;
    return theme.typography.text.lineHeight;
}, function (_a) {
    var theme = _a.theme;
    return theme.typography.text.textAlign;
}, function (_a) {
    var theme = _a.theme;
    return theme.typography.text.textTransform;
}, function (_a) {
    var desktop = _a.desktop, laptop = _a.laptop, laptopL = _a.laptopL, mobileL = _a.mobileL, mobileM = _a.mobileM, mobileS = _a.mobileS, tablet = _a.tablet;
    return util_functions_1.responsiveComponent({
        desktop: desktop,
        laptop: laptop,
        laptopL: laptopL,
        mobileL: mobileL,
        mobileM: mobileM,
        mobileS: mobileS,
        tablet: tablet,
    });
});
exports.TextInputWithFieldLabel = field_label_1.withFieldLabel(exports.TextInput);
var templateObject_1;
