// @react-pakistan/util-functions/storybook/preview.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.globalTypes = exports.parameters = void 0;
const responsive_screens_1 = require("./responsive-screens");
exports.parameters = {
    actions: {
        argTypesRegex: '^on[A-Z].*',
    },
    backgrounds: {
        values: [
            {
                name: 'twitter',
                value: '#00aced',
            },
            {
                name: 'facebook',
                value: '#3b5998',
            },
        ],
    },
    controls: { expanded: true },
    viewport: {
        viewports: Object.assign({}, responsive_screens_1.responsiveScreens),
    },
};
exports.globalTypes = {
    theme: {
        name: 'Theme',
        description: 'Global theme for components',
        defaultValue: 'light',
        toolbar: {
            icon: 'circlehollow',
            items: [
                'light',
                'dark',
            ],
        },
    },
    locale: {
        name: 'Locale',
        description: 'Internationalization locale',
        defaultValue: 'en',
        toolbar: {
            icon: 'globe',
            items: [
                { value: 'en', right: '🇺🇸', title: 'English' },
                { value: 'fr', right: '🇫🇷', title: 'Français' },
                { value: 'es', right: '🇪🇸', title: 'Español' },
                { value: 'zh', right: '🇨🇳', title: '中文' },
                { value: 'kr', right: '🇰🇷', title: '한국어' },
            ],
        },
    },
};
