// @react-pakistan/util-functions/storybook/react-logo-story-template.js
"use strict";
/**
 * A util function to print React Story for Logos
 *
 * @param category string
 * @param logoListDirName string
 * @returns an auto generated icon story file
 * @author Taimoor Khan
 * @remarks React Logo Collection
 */
/** End file docs */
Object.defineProperty(exports, "__esModule", { value: true });
exports.reactLogoStoryTemplate = void 0;
exports.reactLogoStoryTemplate = (category, logoListDirName) => `/*
 * Generated by scripts/generate-logos.js
 * DO NOT EDIT!
 */

import React, { ReactElement } from 'react';
import { LogoWrapper, LogoTextWrapper, LogoItem } from '../styled';
import { StyledStory } from '../styled-app';
import { LOGO_LIST } from './${logoListDirName}';

const logoStyle = { width: '7rem', height: '7rem' };

const renderItem = (
  item : { logo : React.SFC<React.SVGProps<SVGSVGElement>> , name : string }
) : ReactElement => {
  const Logo = item.logo;
  return (
    <LogoItem
      key={item.name}
    >
      <Logo style={logoStyle} />
      <LogoTextWrapper>
        {item.name}
      </LogoTextWrapper>
    </LogoItem>
  );
};

const logos = LOGO_LIST.map(renderItem);

export const ${category} = () : ReactElement => (
  <StyledStory>
    <LogoWrapper>
      {logos}
    </LogoWrapper>
  </StyledStory>
);

export default {
  title: 'Logo/${category}',

  parameters: {
    component: ${category},
  },
};
`;
