// @react-pakistan/util-functions/storybook/responsive-screens.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.responsiveScreens = void 0;
exports.responsiveScreens = {
    mobileSMin: {
        name: 'MobileSMin',
        styles: {
            height: '1024px',
            width: '320px',
        },
        type: 'mobile',
    },
    mobileSMax: {
        name: 'MobileSMax',
        styles: {
            height: '1024px',
            width: '374px',
        },
        type: 'mobile',
    },
    mobileMMin: {
        name: 'MobileMMin',
        styles: {
            height: '1024px',
            width: '375px',
        },
        type: 'mobile',
    },
    mobileMMax: {
        name: 'MobileMMax',
        styles: {
            height: '1024px',
            width: '424px',
        },
        type: 'mobile',
    },
    mobileLMin: {
        name: 'MobileLMin',
        styles: {
            height: '1024px',
            width: '425px',
        },
        type: 'mobile',
    },
    mobileLMax: {
        name: 'MobileLMax',
        styles: {
            height: '1024px',
            width: '767px',
        },
        type: 'mobile',
    },
    tabletMin: {
        name: 'TabletMin',
        styles: {
            height: '100%',
            width: '768px',
        },
        type: 'mobile',
    },
    tabletMax: {
        name: 'TabletMax',
        styles: {
            height: '100%',
            width: '1023px',
        },
        type: 'mobile',
    },
    laptopMin: {
        name: 'LaptopMin',
        styles: {
            height: '100%',
            width: '1024',
        },
        type: 'mobile',
    },
    laptopMax: {
        name: 'LaptopMax',
        styles: {
            height: '100%',
            width: '1439px',
        },
        type: 'mobile',
    },
    laptopLMin: {
        name: 'LaptopLMin',
        styles: {
            height: '100%',
            width: '1440px',
        },
        type: 'mobile',
    },
    laptopLMax: {
        name: 'LaptopLMax',
        styles: {
            height: '100%',
            width: '2559px',
        },
        type: 'mobile',
    },
    desktop: {
        name: 'Desktop',
        styles: {
            height: '100%',
            width: '2560px',
        },
        type: 'mobile',
    },
};
