// @react-pakistan/react-commons-collection/background-image/background-image.js
"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BackgroundImage = void 0;
var util_functions_1 = require("@react-pakistan/util-functions");
var styled_components_1 = __importDefault(require("styled-components"));
exports.BackgroundImage = styled_components_1.default.div(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n  margin: ", ";\n  padding: ", ";\n  width: ", ";\n  height: ", ";\n  background-color: ", ";\n  background-image: ", ";\n  background-repeat: ", ";\n  background-position: ", ";\n  background-size: ", ";\n  border-radius: ", ";\n\n  ", ";\n"], ["\n  margin: ",
    ";\n  padding: ",
    ";\n  width: ",
    ";\n  height: ",
    ";\n  background-color: ",
    ";\n  background-image: ",
    ";\n  background-repeat: ",
    ";\n  background-position: ",
    ";\n  background-size: ",
    ";\n  border-radius: ", ";\n\n  ",
    ";\n"])), function (_a) {
    var imageBackgroundMargin = _a.imageBackgroundMargin;
    return imageBackgroundMargin || '0';
}, function (_a) {
    var imageBackgroundPadding = _a.imageBackgroundPadding;
    return imageBackgroundPadding || '0';
}, function (_a) {
    var imageBackgroundWidth = _a.imageBackgroundWidth;
    return imageBackgroundWidth || '600px';
}, function (_a) {
    var imageBackgroundHeight = _a.imageBackgroundHeight;
    return imageBackgroundHeight || '400px';
}, function (_a) {
    var imageBackgroundColor = _a.imageBackgroundColor, theme = _a.theme;
    return imageBackgroundColor || theme.colors.lightGrey;
}, function (_a) {
    var imageBackgroundUrl = _a.imageBackgroundUrl;
    return "url(" + imageBackgroundUrl + ")";
}, function (_a) {
    var imageBackgroundRepeat = _a.imageBackgroundRepeat;
    return imageBackgroundRepeat || 'no-repeat';
}, function (_a) {
    var imageBackgroundPosition = _a.imageBackgroundPosition;
    return imageBackgroundPosition || 'center center';
}, function (_a) {
    var imageBackgroundSize = _a.imageBackgroundSize;
    return imageBackgroundSize || 'cover';
}, function (_a) {
    var imageBorderRadius = _a.imageBorderRadius;
    return imageBorderRadius || '0';
}, function (_a) {
    var desktop = _a.desktop, laptop = _a.laptop, laptopL = _a.laptopL, mobileL = _a.mobileL, mobileM = _a.mobileM, mobileS = _a.mobileS, tablet = _a.tablet;
    return util_functions_1.responsiveComponent({
        desktop: desktop,
        laptop: laptop,
        laptopL: laptopL,
        mobileL: mobileL,
        mobileM: mobileM,
        mobileS: mobileS,
        tablet: tablet,
    });
});
var templateObject_1;
