// @react-pakistan/react-commons-collection/child-grid/child-grid.js
"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChildGrid = void 0;
var util_functions_1 = require("@react-pakistan/util-functions");
var styled_components_1 = __importDefault(require("styled-components"));
exports.ChildGrid = styled_components_1.default.div(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n  align-self: ", ";\n  grid-column: ", ";\n  grid-row: ", ";\n  justify-self: ", ";\n\n  ", ";\n"], ["\n  align-self: ",
    ";\n  grid-column: ",
    ";\n  grid-row: ",
    ";\n  justify-self: ",
    ";\n\n  ",
    ";\n"])), function (_a) {
    var alignSelf = _a.alignSelf;
    return alignSelf || 'start';
}, function (_a) {
    var gridColumnStart = _a.gridColumnStart, gridColumnEnd = _a.gridColumnEnd;
    return gridColumnStart + " / " + gridColumnEnd;
}, function (_a) {
    var gridRowStart = _a.gridRowStart, gridRowEnd = _a.gridRowEnd;
    return gridRowStart + " / " + gridRowEnd;
}, function (_a) {
    var justifySelf = _a.justifySelf;
    return justifySelf || 'start';
}, function (_a) {
    var desktop = _a.desktop, laptop = _a.laptop, laptopL = _a.laptopL, mobileL = _a.mobileL, mobileM = _a.mobileM, mobileS = _a.mobileS, tablet = _a.tablet;
    return util_functions_1.responsiveComponent({
        desktop: desktop,
        laptop: laptop,
        laptopL: laptopL,
        mobileL: mobileL,
        mobileM: mobileM,
        mobileS: mobileS,
        tablet: tablet,
    });
});
var templateObject_1;
